app/registries/foreman/access_permissions.rb

```diff
@@ -720,4 +720,8 @@
     map.permission :revoke_personal_access_tokens,
                    :"api/v2/personal_access_tokens" => [:destroy]
   end
+
+  permission_set.security_block :table_preferences do |map|
+    map.permission :access_table_preferences, {:"api/v2/table_preferences" => [:index, :create, :destroy, :show, :update]}
```

Comment:
I asked before, but do we need this permission at all? Shouldn't it be just an aspect of the user's preferences, similar to how any user can edit their dashboard layout?

Reply:
We don't have an "access" permission anywhere else. Why not go for `view`, `create`, `destroy`?

Reply:
I would recommend looking at https://github.com/theforeman/foreman/blob/develop/app/registries/foreman/access_permissions.rb#L643-L647
```
  permission_set.security_block :dashboard do |map|
    map.permission :access_dashboard, {:dashboard => [:index, :show, :save_positions, :reset_default, :create, :destroy],
                                      :"api/v2/dashboard" => [:index]
    }
  end
```

What I have is no different for table_preferences.

Reply:
Fair enough, I can see why you chose the access permission. However I still the crud permissions are safer for future development. Just treat this as a hint, I'm ok either way.

Reply:
`access_dashboard` permission limits access to the dashboard since it contains a lot of information you might not want a user to see. Even there, if you are allowed to access it, there is no authorization check for the layout setting. 
In the case of table preferences, I don't see any reason to limit a user's access to the preferences themselves, just like we don't limit the ability to sort or search - it's just a UI preference.

Reply:
True, however I would always grant the user permission to change it's own settings like we do it for personal access tokens. I would use the crud permissions for an admin then who is supposed to fix problems like "I don't see the xyz column anymore. Can you help me get it back?". Just my 2 cents.

Reply:
I think we could add that as a second step if needed, possibly even without using permissions (just allow only admin and self). adding all the permissions checks atm is just complicating the initial implementation - right now we can't merge this since it will allow any user to edit any other user's preference.

Reply:
Sorry I replied via email and it was not aligned to this conversation. Long story short, @tbrisker I think the permission is needed because of tests that verified every action has some permission that covers it. It forces to work only with users preferences thanks to https://github.com/theforeman/foreman/pull/5262/files#diff-6e3452c581d814d25d1c8ac0001adf4dR5 (same thing is used for personal tokens and ssh keys). I don't think we need to change it now. If we're afraid admins can't fix that, let's implement sudo, ideally out of this PR.